ksqldb-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceAsProperties.java

```diff
@@ -154,6 +155,22 @@ public Map<String, String> getKeyFormatProperties(
       builder.put(ProtobufProperties.UNWRAP_PRIMITIVES, ProtobufProperties.UNWRAP);
     }
 
+    final String nullableRep = props.getString(
```

Comment:
This property should only be set if the format is PROTOBUF, right? Looks like this belongs inside the if-block on line 154.

I'm surprised the existing code doesn't have validation to throw if unknown properties are passed (e.g., if this protobuf-specific property is passed to JSON format). Probably out of scope to add this more generally, though we can consider adding validation to reject statements with `PROTOBUF_NULLABLE_REPRESENTATION` set for non-PROTOBUF formats upfront.

Reply:
Done